src/lib/units/year.js

```diff
@@ -36,6 +48,9 @@ addParseToken('YYYY', function (input, array) {
 addParseToken('YY', function (input, array) {
     array[YEAR] = hooks.parseTwoDigitYear(input);
 });
+addParseToken('Y', function (input, array) {
+    array[YEAR] = parseInt(input, 10);
```

Comment:
`parseInt` is the canonical way to do this. I don't want to act as an uglifier, the code is ugly enough already :)